torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2063,6 +2063,14 @@ communication mechanism.
           .def("rank", &::c10d::ProcessGroup::getRank, R"(Get the rank of this process group.)")
           .def("size", &::c10d::ProcessGroup::getSize, R"(Get the size of this process group.)")
           .def("name", &::c10d::ProcessGroup::getBackendName, R"(Get the name of this process group.)")
+          .def(
+              "split_group",
+              &::c10d::ProcessGroup::splitGroup,
+              py::arg("ranks"),
+              py::arg("timeout") = std::nullopt,
+              py::arg("opts") = std::nullopt,
+              py::arg("groupDesc") = std::nullopt,
+              py::call_guard<py::gil_scoped_release>())
```

Comment:
do we also need to add this to PyProcessGroup?

Reply:
oh I almost forgot about it... Sure, let me add it.